app/controllers/teams_controller.rb

```diff
@@ -59,55 +55,75 @@ def destroy
 
   # Assigns team ownership to a specific team member
   def assign_ownership
-    unless owners_include_user
-      team_owners << user
-      redirect_to @team
-    end
+    assign_ownership_to_member
+    redirect_to @team
   end
 
   def delete_ownership
-    delete_owner_if_existing
+    delete_ownership_from_member
     redirect_to @team
   end
 
   def delete_membership
-    delete_owner_if_existing
-    @team.members.delete(user)
+    delete_membership_from_member
     redirect_to @team
   end
 
-  def current_ability
-    @current_ability ||= Ability.new(current_user, params[:team_member])
+  def perform_action_on_multiple_members
+
+    target_members = params[:members]
+    if target_members.include? current_user.id.to_s
+      target_members = (target_members - [current_user.id.to_s]) + [current_user.id.to_s]
+    end
+
+    target_members.each do |member|
+      params[:team_member] = member
+      if params[:assign_ownership]
+        assign_ownership_to_member
+      elsif params[:delete_ownership]
+        delete_ownership_from_member
+      elsif params[:delete_membership]
+        delete_membership_from_member
+      end
+    end
+    redirect_to @team
   end
 
   private
-    def delete_owner_if_existing
-      if owners_include_user
-        team_owners.delete user
+    def assign_ownership_to_member
+      authorize! :assign_ownership, @team
+      member_to_become_owner = TeamUser.find_by(user_id: user, team_id: @team.id)
+      unless member_to_become_owner.nil?
+        member_to_become_owner.assign_ownership
       end
     end
 
+    def delete_ownership_from_member
```

Comment:
For readability, maybe change the way this method retrieves the member who should be deleted: Instead of using the `user` method, pass it as argument to this method.